strongbox-security/strongbox-user-management/src/main/java/org/carlspring/strongbox/gremlin/adapters/UserAdapter.java

```diff
@@ -67,6 +86,42 @@ private User map(Traverser<Map<String, Object>> t)
 
     @Override
     public UnfoldEntityTraversal<Vertex, Vertex> unfold(User entity)
+    {
+        String storedUserId = Vertices.USER + ":" + UUID.randomUUID().toString();
+
+        EntityTraversal<Vertex, Vertex> userRoleTraversal = __.identity();
+        EntityTraversal<Vertex, Vertex> unfoldTraversal = __.identity();
+
+        if (entity.getNativeId() != null)
+        {
+            unfoldTraversal.V(entity.getNativeId())
+                           .sideEffect(__.outE(Edges.USER_HAS_USER_ROLES).drop());
+        }
+
+        for (UserRole userRole : entity.getRoles())
+        {
+            userRoleTraversal = userRoleTraversal.V(userRole)
+                                                 .saveV(userRole.getUuid(),
+                                                        userRoleAdapter.unfold(userRole));
+
+            userRoleTraversal = userRoleTraversal.addE(Edges.USER_HAS_USER_ROLES)
+                                                 .from(__.<Vertex, Vertex>select(storedUserId).unfold())
+                                                 .inV();
+
+            userRoleTraversal = userRoleTraversal.inE(Edges.USER_HAS_USER_ROLES).outV();
+
+        }
+
+        unfoldTraversal = unfoldTraversal.map(unfoldUser(entity))
+                                         .store(storedUserId)
+                                         .flatMap(userRoleTraversal)
+                                         .fold()
```

Comment:
we don't need `fold` and above `map` here because `userRoleTraversal` will end with user vertex.

Reply:
Fixed.